File: JAVA DSA/InsertionSort.java
import java.util.Arrays;

public class InsertionSort {
    public static void main(String[] args) {
        int[] arrToSort = {6,7,8,9,0,3,4,2};
        insertionSort(arrToSort);
        System.out.println(Arrays.toString(arrToSort));
    }
    static void insertionSort(int[] arr){
        for (int i = 0; i < arr.length-1; i++) {
                int k = i+1;
                while(k > 0){
                    if (arr[k] < arr[k - 1]) {
                        int temp = arr[k];
                        arr[k] = arr[k - 1];
                        arr[k - 1] = temp;
                    }else{
                        break;
                    }
                    k--;
                }
            System.out.println(Arrays.toString(arr));
         }
        }
    }

File: SearchingAndSorting/insertionSort.js
function insertionSort(array) {
    for (let i = 1; i < array.length; i++) {
      let currentValue = array[i];
      let j = i-1;
      while(j >= 0 && array[j] > currentValue) {
        array[j + 1] = array[j];
        j--;
      }
      array[j + 1] = currentValue;
    }
    return array;
  }

// TC O(N^2)
// SC O(N)

insertionSort([5,6,7,2,8,9,1]);